crates/chain/src/tx_graph.rs

```diff
@@ -1110,6 +1122,118 @@ impl<A> ForEachTxOut for TxGraph<A> {
     }
 }
 
+/// An iterator that traverses transaction ancestors to a specified depth.
+///
+/// This `struct` is created by the [`walk_ancestors`] method of [`TxGraph`].
+///
+/// [`walk_ancestors`]: TxGraph::walk_ancestors
+pub struct TxAncestors<'g, A> {
+    graph: &'g TxGraph<A>,
+    visited: HashSet<Txid>,
+    stack: Vec<(usize, Transaction)>,
```

Comment:
```suggestion
    stack: Vec<(usize, &'g Transaction)>,
```

The iterator should use references of transactions instead of cloning them. You'll need to play around with lifetimes to make this happen.